api/src/test/java/org/openmrs/test/BaseContextSensitiveTest.java

```diff
@@ -996,4 +998,20 @@ public void skipBaseSetup() throws Exception {
 		skipBaseSetup = true;
 	}
 	
-}
+	/**
+	 * Instance variable used by the {@link #baseSetupWithStandardDataAndAuthentication()} method to
+	 * know whether the current "@Test" method has asked to be _not_ do the authentication.
+	 * 
+	 * @see SkipBaseSetup
+	 * @see SkipBaseSetupAnnotationExecutionListener
+	 * @see #baseSetupWithStandardDataAndAuthentication()
+	 */
+	private boolean skipAuthenticationInBaseSetup = false;
```

Comment:
@mks-d i would not add this in anticipation of future use cases. 😊 
In otherwords, i have not seen any part of your changes in the pull request that required any change in this class.

Reply:
@dkayiwa it is necessary as soon as you want to test an alternate auth scheme, have a look at `AlternateAuthenticationSchemeContextTest`.

This test wires a test scheme that requires its own credentials, that scheme is therefore unable to authenticate 'admin' with the OpenMRS credentials as it doesn't know about them and hence what do to with such credentials.

Reply:
I have put a comment in TestUsernameAuthenticationScheme in regards to this.